dashboards-observability/public/components/common/search/autocomplete_logic.ts

```diff
@@ -335,4 +354,85 @@ export const getSuggestions = async (
     }
     return [];
   }
+  return [];
+};
+
+const parseForIndex = (query: string) => {
+  for (let i = 0; i < regexForIndex.length; i++) {
+    const groupArray = regexForIndex[i].exec(query);
+    if (groupArray) {
+      return groupArray[1];
+    }
+  }
+  return '';
+};
+
+const parseForNextSuggestion = (command: string) => {
+  for (let i = 0; i < regexForSuggestion.length; i++) {
+    const groupArray = regexForSuggestion[i].exec(command);
+    if (groupArray) {
+      return regexForSuggestion[i];
+    }
+  }
+};
+
+export const getSuggestionsAfterSource = async (currQuery: string, dslService: DSLService) => {
+  const splitSpaceQuery = currQuery.split(' ');
+  const splitPipeQuery = currQuery.split('|');
+
+  const lastWord = splitSpaceQuery[splitSpaceQuery.length - 1];
+  const lastCommand = splitPipeQuery[splitPipeQuery.length - 1];
+
+  if (lastCommand.split(' ')[0] === 'source') {
+    currIndex = parseForIndex(currQuery);
+    getFields(dslService);
+    currField = '';
+    currFieldType = '';
+    return fillSuggestions(currQuery, lastWord, [{ label: '|' }]);
+  } else {
+    const next = parseForNextSuggestion(lastCommand);
+    switch (next) {
+      case PIPE_AFTER_FIELD:
+        return fillSuggestions(currQuery, lastWord, [
+          { label: '|' },
+          { label: 'keepempty=true' },
+          { label: 'consecutive=true' },
+        ]);
+      case CLOSE_AFTER_DATA:
+        console.log('close after data');
```

Comment:
Minor: do we need logging at here and other branches?

Reply:
Oh I forgot to remove these! Good catch.